### ISSUE TITLE
Problems installing on Windows with Conda

### ISSUE THREAD

**Issue by rsignell-usgs** [titusjan/argos#5]
I tried  installing Argos on Windows using Conda thusly:
```
conda create -n ARGOS python=3 numpy scipy netcdf4 h5py pandas pillow
activate ARGOS
pip install pyqt5
pip install pyqtgraph
pip install argos
```
but running
```
python -m argos.main
```
returns
```
from PyQt5 import QtCore, QtGui, QtWidgets, QtSvg
ImportError: DLL load failed: The specified module could not be found.
```
@ocefpaf tells me this is a known problem with the defaults channel at Continuum.

**Comment by titusjan** [comment]
It seems that installing PyQt with `pip` doesn't work together with Conda. I don't know if this is a temporary issue or permanent. 

Anyway, try uninstalling PyQt with `pip uninstall pyqt5` and then installing it via `conda install pyqt`. This worked for the guy in issue #1.

**Comment by rsignell-usgs** [comment]
Unfortunately conda does not yet have `pyqt5` on conda-forge, but they are working on it:
https://github.com/conda-forge/staged-recipes/pull/2210#issuecomment-273059844

**Comment by bilderbuchi** [comment]
I just installed `pyqt` with conda and that gave me version 5.6.something.

**Comment by ocefpaf** [comment]
Like @rsignell-usgs mentioned above, outside of the `conda-forge` realm, it should work. However, many rely on `conda-forge` to build their envs and get more software than just `argos`.

An alternative, until we fix Qt in conda-forge, is to use an `environment.yml` with the dependencies from `defaults` instead.

**Comment by bilderbuchi** [comment]
Ah, I was not aware that conda-forge replaces/overrides(?) the default conda package list.

**Comment by ocefpaf** [comment]
It does note replace or overrides. It dependents on how you use it. If `conda-forge` is on top of `defaults` in your `.condarc` file (or `environment.yml`) `conda` will prefer `conda-forge` packages. The same is true doe `defaults` if you revert the order.

TL;DR
`conda-forge` is note 100% compatible with `defaults` because `conda-forge` is almost always more up-to-date with the exception of Qt due to CI time constraints. So the channel order matters.

**Comment by bilderbuchi** [comment]
Alright, thanks for clearing this up.

**Comment by rsignell-usgs** [comment]
I tried building on Windows using only the default channel, but the version of pyqt **on Windows** there is still at 4.11.4,  so no go.  Here's what I did:
```
conda env create -f argos_environment.yml
```
with `environment.yml`:
```
name: ARGOS
channels:
  - defaults
dependencies:
  - python=3.5
  - numpy
  - pyqt
  - pyqtgraph
  - h5py
  - netcdf4
  - pillow
  - scipy
  - pandas
```
And listing the pyqt version results:
```
(ARGOS) C:\Users\rsignell\documents\github>conda list pyqt
# packages in environment at C:\Users\rsignell\AppData\Local\Continuum\Miniconda
3\envs\ARGOS:
#
pyqt                      4.11.4                   py35_7    defaults
pyqtgraph                 0.9.10                   py35_1    defaults
```

**Comment by bilderbuchi** [comment]
Try leaving out pyqtgraph and installing that one via pip

**Comment by rsignell-usgs** [comment]
@bilderbuchi , indeed leaving out `pyqtgraph` allowed conda to find a more recent version of `pyqt`:
```
(ARGOS) C:\Users\rsignell\documents\github>conda list pyqt
# packages in environment at C:\Users\rsignell\AppData\Local\Continuum\Miniconda
3\envs\ARGOS:
#
pyqt                      5.6.0                    py35_1    defaults
pyqtgraph                 0.10.0                    <pip>
```
So progress!   

Unfortunately, when I run  `python -m argo.main`, however, I get this:

![2017-01-17_12-18-20](https://cloud.githubusercontent.com/assets/1872600/22031734/9142d488-dcaf-11e6-8293-7e47221de74e.png)

**Comment by rsignell-usgs** [comment]
The above `argos_environment.yml` works great on Linux though.  :stuck_out_tongue:

**Comment by titusjan** [comment]
Is there any output on the console? Perhaps if you start Argos with debug messages. Can you try to start with `python -m argos.main -l debug` please?

**Comment by rsignell-usgs** [comment]
Okay, I tried it, but you aren't going to like it:
![2017-01-18_6-47-26](https://cloud.githubusercontent.com/assets/1872600/22063286/07db0006-dd4c-11e6-907d-47b95451175e.png)

**Comment by titusjan** [comment]
I looked in the code. Directly after main.py:82 Argos will try to import PyQt. So I think there is still an issue with your PyQt installation. Let's try to confirm this by bypassing Argos and just import PyQt from Python. Can you start `python` and then type the following on the Python command prompt: `from PyQt5 import QtCore, QtGui, QtWidgets`. I expect this gives the same error behaviour.

**Comment by rsignell-usgs** [comment]
You are correct!  It gives the same behaviour.
